src/main/java/de/metas/ui/web/attachments/DocumentAttachments.java

```diff
@@ -76,7 +78,8 @@
 	private static final Splitter ID_Splitter = Splitter.on(ID_SEPARATOR);
 	private static final Joiner ID_Joiner = Joiner.on(ID_SEPARATOR);
 
-	private final transient AttachmentEntryService attachmentEntryService = Adempiere.getBean(AttachmentEntryService.class);
+	private final transient AttachmentEntryService attachmentEntryService = SpringContextHolder.instance.getBean(AttachmentEntryService.class);
+	private final transient TableAttachmentListenerRepository tableAttachmentListenerRepository = SpringContextHolder.instance.getBean(TableAttachmentListenerRepository.class);
```

Comment:
consider having those 2 services as ctor params.

Reply:
so we will avoid any injection here, in this non-service-ish object.

Reply:
PS: in DocumentAttachmentsRestController pls get rid of those `@Autowired` beans. Use ctor.